src/components/Comments/CommentItem/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+
+import { Grid } from "@material-ui/core";
+import { Edit as EditIcon, Delete as DeleteIcon } from "@material-ui/icons";
+
+import AvatarBase from "../../Avatars/AvatarBase";
+import BoxBase from "../../Boxs/BoxBase";
+import ButtonBase from "../../Buttons/ButtonBase";
+import DividerBase from "../../Dividers/DividerBase";
+import TypographyBase from "../../Typography/TypographyBase";
+import CommentForm from "../CommentForm";
+
+import convertLocateTime from "src/utils/convertLocateTime";
+
+export interface CommentItemProps {
+    commentId: string;
+    authorName: string;
+    authorAvatar?: string | null;
+    content: string;
+    createdAt: string;
+    deleteCommentHandler: (props: { commentId: string }) => void;
+    isDeleteCommentLoading?: boolean;
+    updateCommentHandler: (props: { commentId: string; content: string }) => void;
+    isUpdateCommentLoading?: boolean;
+}
+
+export const defaultAvatarSrc =
+    "https://www.gravatar.com/avatar/d7465b8fce04fd17a376618ba1eb8ddc.jpg?d=https%3A//d2mcnjhkvrfuy2.cloudfront.net/static/images/avatar_default.png&s=64";
+
+const CommentItem = (props: CommentItemProps) => {
+    const {
+        commentId,
+        authorName,
+        authorAvatar,
+        content,
+        createdAt,
+        deleteCommentHandler,
+        isDeleteCommentLoading,
+        updateCommentHandler,
+        isUpdateCommentLoading,
+    } = props;
+
+    const [isEdit, setIsEdit] = useState(false);
+
+    const deleteHandler = () => {
+        deleteCommentHandler({ commentId });
+    };
+    const updateHandler = (content: string) => {
+        updateCommentHandler({ commentId, content });
+        setIsEdit(false);
+    };
+
+    const createdTime = convertLocateTime(new Date(createdAt));
+    const commentCreatedTime =
+        createdTime.toLocaleString("en-US", { dateStyle: "short" }) +
+        " AT " +
+        createdTime.toLocaleString("en-US", { hour: "2-digit", minute: "2-digit" });

```

Comment:
I think we should export this as a reusable function

Reply:
Okay I have just made it a function ^^